components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/builder/AuthReqBuilder.java

```diff
@@ -106,4 +185,150 @@ private static NameIDPolicy buildNameIDPolicy() {
+                org.opensaml.xml.signature.X509Certificate cert =
+                        (org.opensaml.xml.signature.X509Certificate) buildXMLObject(
+                                org.opensaml.xml.signature.X509Certificate.DEFAULT_ELEMENT_NAME);
+                String value = Base64.encodeBytes(cred.getEntityCertificate().getEncoded());
+                cert.setValue(value);
+                data.getX509Certificates().add(cert);
+                keyInfo.getX509Datas().add(data);
+                signature.setKeyInfo(keyInfo);
+            } catch (CertificateEncodingException e) {
+                throw new Exception("Error getting certificate", e);
+            }
+
+            authnRequest.setSignature(signature);
+
+            List<Signature> signatureList = new ArrayList<Signature>();
+            signatureList.add(signature);
+
+            // Marshall and Sign
+            MarshallerFactory marshallerFactory =
+                    org.opensaml.xml.Configuration.getMarshallerFactory();
+            Marshaller marshaller = marshallerFactory.getMarshaller(authnRequest);
+
+            marshaller.marshall(authnRequest);
+            
+            Signer.signObjects(signatureList);
+            return authnRequest;
+
+        } catch (Exception e) {
+            throw new Exception("Error while signing the SAML Request message", e);
+        }
+    }
+    
+
+    /**
+     * Sign the SAML AuthnRequest message
+     *
+     * @param logoutRequest
+     * @param signatureAlgorithm
+     * @param cred
+     * @return
+     */
+    
+    
+    public static LogoutRequest setSignature(LogoutRequest logoutRequest, String signatureAlgorithm,
+            X509Credential cred) throws Exception {
+        try {
+            Signature signature = (Signature) buildXMLObject(Signature.DEFAULT_ELEMENT_NAME);
+            signature.setSigningCredential(cred);
+            signature.setSignatureAlgorithm(signatureAlgorithm);
+            signature.setCanonicalizationAlgorithm(SignatureConstants.ALGO_ID_C14N_EXCL_OMIT_COMMENTS);
+
+            try {
+                KeyInfo keyInfo = (KeyInfo) buildXMLObject(KeyInfo.DEFAULT_ELEMENT_NAME);
+                X509Data data = (X509Data) buildXMLObject(X509Data.DEFAULT_ELEMENT_NAME);
+                org.opensaml.xml.signature.X509Certificate cert =
+                        (org.opensaml.xml.signature.X509Certificate) buildXMLObject(
+                                org.opensaml.xml.signature.X509Certificate.DEFAULT_ELEMENT_NAME);
+                String value = Base64.encodeBytes(cred.getEntityCertificate().getEncoded());
+                cert.setValue(value);
+                data.getX509Certificates().add(cert);
+                keyInfo.getX509Datas().add(data);
+                signature.setKeyInfo(keyInfo);
+            } catch (CertificateEncodingException e) {
+                throw new SecurityException("Error getting certificate", e);
```

Comment:
Unnecessary try-cache block removed in #849. Exception handling improved in #849, #1523 (a3c032a) and #1528 (c6d6cb2)